### DISCUSSION TITLE
v5 session expires rotation

### DISCUSSION THREAD

**Discussion by jclepin** [nextauthjs/next-auth#10807] [Help]
Hi,
I use Next.js 14 & Auth.js 5

I want to check if session is about to expire to ask user if he/she is still here (reading, talking, watching a vidéo ...)
But, when I SWR the api route and get the token with auth(), the function automatically rotates the expiry date so it will never expire.

How should I do to get the expires value of the current token without renewing it ?

Regards

**Comment by jclepin** [comment]
So, no way to know when the token is about to expire ?

**Reply by william-will-angi** [reply]
The easiest way to accomplish this is probably to add the expiration to your next auth session via the `jwt` callback and then expose that expiry to your session endpoint in the `session` callback.

https://next-auth.js.org/configuration/callbacks#jwt-callback

**Reply by jclepin** [reply]
Hi William,

Thanks for stopping by.

The problem with V5 is that the new auth() method (which is now the only method) updates the expiry date automatically.
So I need to be able to retrieve the expiration date in real time (hence the SWR library to call an api route that gives me the jwt) but without calling auth().
Or to be able to call auth() without updating the expiration date. Which I think would be the most useful.

Reminder: We're client side. My aim is to warn the user (toast) near the end of the session in case he's watching a video or reading an article ... If the user doesn't respond, I close the session, otherwise I extend it (with a call to auth()).

**Reply by whispernight** [reply]
any news on this? I'm, stuck with a similar issue.
I want to log out the user if the session expires, but it keeps getting renewed

**Reply by hjoelh** [reply]
@william-will-angi is right, here an article on how to do it https://hjoel.xyz/next-auth-fixed-session @whispernight

**Comment by chember004** [comment]
Hi guys, this might help.

for SSR, you can use auth()
If CSR then you can use useSession(). Don't forget the AuthWrapper provider
both of them have an update method.

for SSR, better you create a call to reuse it or just do it directly.
![image](https://github.com/user-attachments/assets/66ee12cb-b012-419c-addc-a1e467fd53fb)

for CSR, directly call the useSession(). 
![image](https://github.com/user-attachments/assets/3c39ad93-ce8a-4e52-8d19-0fbdc4de431e)

Both of them has update

for token expiration
https://medium.com/@yusufs35/integration-of-nextjs14-next-auth5-and-an-external-authentication-api-1f35a9db0eed
https://github.com/nextauthjs/next-auth/discussions/9595

Hope these helps HEHE
